lib/checkout_sdk/api_resource.rb

```diff
@@ -51,6 +51,15 @@ def post_request(path, data)
     )
   end
 
+  def post_request_public(path, data)
+    checkout_connection.post(
+      path: path,
+      body: MultiJson.dump(delete_blank(data)),
+      headers: { "Content-Type" => "application/json",
+                 "Authorization" => "#{CheckoutSdk.configuration.public_key}" }
```

Comment:
Instead of repeating the code I would create a private conditional key method that returns the public_key incase the path is "/tokens"